### PR TITLE
Tab component v1

### PR DESCRIPTION
**Pull request by saurabhsutar192**
Tab Component

### CODE REVIEW

**Comment by pruthvi2103** [comment]
pl fill the PR template too 🥲

File: lib/src/components/Tab/Tab.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                  shape={"rounded"}
+                  color={"#DA2C2C"}
+                />
+              )}
+            </div>
+          );
+        })}
+      </div>
+      <div className={`${contentStyles}`}>{children(selectedTab)}</div>
+    </div>
```
Comment (pruthvi2103) [inline lib/src/components/Tab/Tab.tsx:84]: Also forward Ref the tab, refer any of the components to see how we do it

File: lib/src/components/Tab/Tab.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+export const Tab = ({
+  defaultValue,
+  value,
+  color = "#2F80ED",
+  background = "#d7e9ff",
+  onChange = () => {},
+  grow = false,
+  height = "40px",
+  tabs,
```
Comment (pruthvi2103) [inline lib/src/components/Tab/Tab.tsx:22]: better to name this better, something like tabData

File: lib/src/components/Tab/Tab.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  children,
+  ...nativeProps
+}: TabProps) => {
+  const [selectedTab, setSelectedTab] = useState<TabsObjectProps>(
+    {} as TabsObjectProps
+  );
+
+  useEffect(() => {
+    const comparator = value || defaultValue;
+    const tab = tabs.find((arr) => arr.value === comparator);
```
Comment (pruthvi2103) [inline lib/src/components/Tab/Tab.tsx:34]: `arr` is a generic name, name this to something like `tabDataItem`

File: lib/src/components/Tab/Tab.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  className,
+  children,
+  ...nativeProps
+}: TabProps) => {
+  const [selectedTab, setSelectedTab] = useState<TabsObjectProps>(
+    {} as TabsObjectProps
+  );
+
+  useEffect(() => {
+    const comparator = value || defaultValue;
```
Comment (pruthvi2103) [inline lib/src/components/Tab/Tab.tsx:33]: `comparator` also does not tell the purpose, could we name this as `selectedTabValue`

File: lib/src/components/Tab/tab.css.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  color: "#2F80ED",
+  background: "#BDDBFF60",
+  height: "50px",
+});
+
+export const tabHeaderContainerStyles = style({
+  height: tabVars.height,
+  position: "relative",
+  display: "flex",
+  alignItems: "center",
```
Comment (pruthvi2103) [inline lib/src/components/Tab/tab.css.ts:15]: Better to have used the `<Flex>` component gives us confidence when using our own components and we can find out potential flaws, keep this as least priority, can be ignored for this version of tabs

File: lib/src/components/Tab/Tab.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+  return (
+    <div>
+      <div
+        style={assignInlineVars({
+          [tabVars.height]: height,
+        })}
+        className={`${tabHeaderContainerStyles}`}
+      >
+        {tabs.map((arr, ind) => {
```
Comment (pruthvi2103) [inline lib/src/components/Tab/Tab.tsx:46]: same as above `arr` is generic name. in the future the purpose will be difficult to understand as complexity of code increases

File: lib/src/components/Tab/Tab.tsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                  [tabVars.color]: color,
+                  [tabVars.background]: background,
+                }),
+                ...style,
+              }}
+              className={`${tabClass} ${className}`}
+              onClick={(event) => {
+                onChange(arr, event);
+                !value && setSelectedTab(arr);
+              }}
```
Comment (pruthvi2103) [inline lib/src/components/Tab/Tab.tsx:65]: create a function above like `internalOnClickHandler` and then do `onClick={internalOnClickHandler}`, helps keep the JSX part clean

File: lib/src/components/Tab/Tab.tsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+                  className={`${badgeStyles}`}
+                  shape={"rounded"}
+                  color={"#DA2C2C"}
+                />
+              )}
+            </div>
+          );
+        })}
+      </div>
+      <div className={`${contentStyles}`}>{children(selectedTab)}</div>
```
Comment (pruthvi2103) [inline lib/src/components/Tab/Tab.tsx:83]: ooo didnt know about this, we can do method call for children. Nice 💯

File: lib/src/components/Tab/tab.css.ts [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  padding: "8px 0",
+});
+
+export const iconStyles = style({
+  marginRight: "6px",
+});
+
+export const badgeStyles = style({
+  marginLeft: "6px",
+});
```
Comment (pruthvi2103) [inline lib/src/components/Tab/tab.css.ts:76]: awesome, CSS looks really clean and nice 🔥

File: lib/src/components/Tab/tab.css.ts [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+export const tabRecipe = recipe({
+  base: {
+    cursor: "pointer",
+    height: "100%",
+    justifyContent: "center",
+    position: "relative",
+    zIndex: 1,
+    padding: "8px 10px",
+    display: "flex",
+    alignItems: "center",
```
Comment (pruthvi2103) [inline lib/src/components/Tab/tab.css.ts:28]: again could've used the `<Flex>` component here, but its fine for this version

**Review by pruthvi2103 — CHANGES_REQUESTED** [review]
Some minor nitpicks, apart from that the component looks Awesome!!! 

there are items that can be taken up for v2 of tabs, please create an issue of the things we plan to pickup when we develop v2 of tabs. Also one of the main things for v2 should be accessibility (the tabs cant be used with keyboard navigation)

File: lib/src/components/Tab/Tab.tsx (hunk 1 above)
Comment (saurabhsutar192) [reply lib/src/components/Tab/Tab.tsx:84]: Sure

File: lib/src/components/Tab/tab.css.ts (hunk 10 above)
Comment (saurabhsutar192) [reply lib/src/components/Tab/tab.css.ts:28]: wokay

**Comment by pruthvi2103** [comment]
Pl resolve conflicts too

File: lib/src/components/Tab/Tab.tsx (hunk 3 above)
Comment (saurabhsutar192) [reply lib/src/components/Tab/Tab.tsx:34]: ok

**Review by pruthvi2103 — APPROVED** [review]
LGTM! Great Work 💯 

Pl just comment the issue `#`s here too so that we know they are connected

**Pull request merged**